### ISSUE TITLE
Errors when running quick_example.py code

### ISSUE THREAD

**Issue by carrtesy** [hugothimonier/NPT-AD#2]
Hello, 

Thanks for your great work, and I appreciate your efforts in providing code.

I tried running your code, but noticed some difficuties, and seeking for your help.

1. get_preset function
when calling [get_preset](https://github.com/hugothimonier/NPT-AD/blob/c4c4d00aa98540a1254f472f56d5129b0915db9a/npt/config_manager.py#L257 
) function, it makes some errors.
I guess that is because the function is calling [cls.from_dict()](https://github.com/hugothimonier/NPT-AD/blob/c4c4d00aa98540a1254f472f56d5129b0915db9a/npt/config_manager.py#L287), with argument as string type.
Proper way I guess, is to read as json first and pass that object, but I am not sure.

```python
@classmethod
def _get_small_dataset_preset(cls) -> 'NPTADConfig':
    """Configuration optimized for small datasets."""
    config_dict = "./config/base/base_small_dataset.json"
    config = cls.from_dict(config_dict)
    return config

@classmethod
def _get_small_dataset_high_d_preset(cls) -> 'NPTADConfig':
    """Configuration optimized for small datasets."""
    config_dict = "./config/base/base_small_dataset_high_nb_features.json"
    config = cls.from_dict(config_dict)
    return config

@classmethod
def _get_medium_dataset_preset(cls) -> 'NPTADConfig':
    """Configuration optimized for small datasets."""
    config_dict = "./config/base/base_medium_dataset.json"
    config = cls.from_dict(config_dict)
    return config

@classmethod
def _get_medium_dataset_high_d_preset(cls) -> 'NPTADConfig':
    """Configuration optimized for small datasets."""
    config_dict = "./config/base/base_medium_dataset_high_nb_features.json"
    config = cls.from_dict(config_dict)
    return config

@classmethod
def _get_large_dataset_preset(cls) -> 'NPTADConfig':
    """Configuration optimized for small datasets."""
    config_dict = "./config/base/base_large_dataset.json"
    config = cls.from_dict(config_dict)
    return config

@classmethod
def _get_large_dataset_high_d_preset(cls) -> 'NPTADConfig':
    """Configuration optimized for small datasets."""
    config_dict = "./config/base/base_large_dataset_high_nb_features.json"
    config = cls.from_dict(config_dict)
    return config
```

2. infinite loop
When running code [gen_mask_matrices](https://github.com/hugothimonier/NPT-AD/blob/c4c4d00aa98540a1254f472f56d5129b0915db9a/npt/mask.py#L301) function in mask.py, there's an infinite loop.

```python
  mask_prob = config.model.augmentation_bert_mask_prob['val']
  std = np.sqrt(Nm * mask_prob * (1 - mask_prob))

  # This gives the total number of masks sampled in our approximative
  # sampling scheme as.
  # maske sure all masks are different
  while ( (num_masks==0) or (num_masks<=recon) ):
      num_masks_sampled = 0

      while num_masks_sampled < 1:
          num_masks_sampled = int(
              mask_prob * Nm +
              np.random.normal(0, std))
```

This is problematic as mask_prob is set to 0, as ModelConfig() sets that as 0 [link](https://github.com/hugothimonier/NPT-AD/blob/c4c4d00aa98540a1254f472f56d5129b0915db9a/npt/config_manager.py#L40).

```python
augmentation_bert_mask_prob: Dict[str, float] = field(default_factory=lambda: {
      'train': 0.15, 'val': 0.0, 'test': 0.0
  })
```

Could you show me right way to resove these issues?
And I wanna run your code as simple as possible to investigate your paper properly, and could you give me a way to run your code as simple as possible?

Sincerely,
Carrtesy

**Comment by hugothimonier** [comment]
Hello @carrtesy,

Thanks a lot for reporting this issue! I've investigated your points and implemented the fixes:

1.  **Loading the Preset:** You were absolutely right. I was calling the wrong method. I've updated `NPTADConfig` to correctly use `from_json` as you suggested.

2.  **`gen_mask_matrices()` and Default Config:** This was a more complex bug:
    * The primary fix was realizing I had **incorrectly set** `data/deterministic_masks: false` for *all* presets, causing the `ModelConfig.augmentation_bert_mask_prob` default of **`0.0`** to cause errors. This is now corrected.
    * To prevent this specific issue from happening again if a user *intentionally* sets `data/deterministic_masks: false`, I've also modified the default value for `ModelConfig.augmentation_bert_mask_prob['val']`.

`examples/quick_start.py` should now run smoothly!

---

Please try running the code and confirm the fix. If everything looks good, I'll close this issue.

**Comment by carrtesy** [comment]
Thanks a lot for your answer.
I executed the code with the following:
```sh
python examples/quick_start.py
```
It seems to run, as I see the log "running training epoch: #num".

But I encountered a minor issue here:
```sh
1. Running quick test on separable dataset...
Using GPU: cuda:0
Loading dataset: separable
Quick test failed: [Errno 2] No such file or directory: './data/abalone/separable.npy'
This is expected if the dataset is not available.

2. Training with custom configuration...
Using GPU: cuda:0
Loading dataset: abalone
Custom training failed: [Errno 2] No such file or directory: './data/abalone/abalone.data'
This is expected if the dataset is not available.
```

I'm not sure, but It seems that the code is refering ./data/abalone dir as base directory for the dataset.

Can I ask a valid way to set this path?

Here is the full log when running the code. 

```sh

NPT-AD Quick Start Example
========================================

1. Running quick test on separable dataset...
Using GPU: cuda:0
Loading dataset: separable
Quick test failed: [Errno 2] No such file or directory: './data/abalone/separable.npy'
This is expected if the dataset is not available.

2. Training with custom configuration...
Using GPU: cuda:0
Loading dataset: abalone
Custom training failed: [Errno 2] No such file or directory: './data/abalone/abalone.data'
This is expected if the dataset is not available.

3. Using configuration presets...
Using GPU: cuda:0
Loading dataset: separable
Forcing data rebuild and recache.
Dataset loaded: 1900 samples, 7 features
Setting up model...
Building NPT.
Clipping gradients to value 1.0.
Model has 249830 parameters,batch size -1.
Initialized "lookahead_lamb" optimizer.
/NPT-AD/npt/utils/model_init_utils.py:33: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  scaler = GradScaler(enabled=config.model.amp)
Initialized gradient scaler for Automatic Mixed Precision.
Model initialized: 249830 parameters
Starting training...
Warming up for 0.0/1000.0 steps.
/opt/conda/envs/nptad/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:216: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn(
Initialized "flat_and_anneal" learning rate scheduler.
running training epoch: 1
/NPT-AD/npt/mask.py:114: UserWarning: torch.sparse.SparseTensor(indices, values, shape, *, device=) is deprecated.  Please use torch.sparse_coo_tensor(indices, values, shape, dtype=, device=). (Triggered internally at ../torch/csrc/utils/tensor_new.cpp:641.)
  mask = torch.sparse.FloatTensor(
/NPT-AD/npt/train.py:533: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast(enabled=self.config.model.amp):
running training epoch: 2
running training epoch: 3
running training epoch: 4
running training epoch: 5
running training epoch: 6
/NPT-AD/npt/utils/optim_utils.py:58: UserWarning: This overload of add_ is deprecated:
        add_(Number alpha, Tensor other)
Consider using one of the following signatures instead:
        add_(Tensor other, *, Number alpha = 1) (Triggered internally at ../torch/csrc/utils/python_arg_parser.cpp:1581.)
  slow.add_(group['lookahead_alpha'], fast_p.data - slow)
running training epoch: 7
running training epoch: 8
running training epoch: 9
running training epoch: 10

```

Thanks again for your help.
Carrtesy
